crates/bevy_text/src/text2d.rs

```diff
@@ -95,47 +95,41 @@ pub fn extract_text2d_sprite(
         .map(|window| window.resolution.scale_factor() as f32)
         .unwrap_or(1.0);
 
-    for (entity, computed_visibility, text, text_layout_info, anchor, text_transform) in
+    for (entity, computed_visibility, text, text_layout_info, anchor, global_transform) in
         text2d_query.iter()
     {
         if !computed_visibility.is_visible() {
             continue;
         }
 
-        let text_glyphs = &text_layout_info.glyphs;
         let text_anchor = anchor.as_vec() * Vec2::new(1., -1.) - 0.5;
-        let alignment_offset = text_layout_info.size * text_anchor;
+        let alignment_translation = text_layout_info.size * text_anchor;
+        let transform = *global_transform
+            * Transform::from_scale(Vec3::splat(scale_factor.recip()))
+            * Transform::from_translation(alignment_translation.extend(0.));
+
         let mut color = Color::WHITE;
         let mut current_section = usize::MAX;
-        for text_glyph in text_glyphs {
-            if text_glyph.section_index != current_section {
-                color = text.sections[text_glyph.section_index]
-                    .style
-                    .color
-                    .as_rgba_linear();
-                current_section = text_glyph.section_index;
+        for PositionedGlyph {
+            position,
+            atlas_info,
+            section_index,
+            ..
+        } in &text_layout_info.glyphs
+        {
+            if *section_index != current_section {
+                color = text.sections[*section_index].style.color.as_rgba_linear();
+                current_section = *section_index;
             }
-            let atlas = texture_atlases
-                .get(&text_glyph.atlas_info.texture_atlas)
-                .unwrap();
-            let handle = atlas.texture.clone_weak();
-            let index = text_glyph.atlas_info.glyph_index;
-            let rect = Some(atlas.textures[index]);
-
-            let glyph_transform =
-                Transform::from_translation((alignment_offset + text_glyph.position).extend(0.));
-
-            let transform = *text_transform
-                * GlobalTransform::from_scale(Vec3::splat(scale_factor.recip()))
-                * glyph_transform;
+            let atlas = texture_atlases.get(&atlas_info.texture_atlas).unwrap();
 
             extracted_sprites.sprites.push(ExtractedSprite {
                 entity,
-                transform,
+                transform: transform * Transform::from_translation(position.extend(0.)),
```

Comment:
Same here. We might want to use `GlobalTransform::from_translation` instead.

Reply:
Yes you are right, using `Transform` is pointless, changed it.